src/OpenTelemetry.Exporter.Geneva/Internal/TableNameSerializer.cs

```diff
@@ -0,0 +1,313 @@
+            {
+                return tableName;
+            }
+
+            byte[] mappedTableName = null;
+
+            // If user configured table name mappings run resolution logic.
+            if (this.m_tableMappings != null
+                && !this.m_tableMappings.TryGetValue(categoryName, out mappedTableName))
+            {
+                // Find best match if an exact match was not found.
+
+                string currentKey = null;
+
+                foreach (var mapping in this.m_tableMappings)
+                {
+                    if (!categoryName.StartsWith(mapping.Key, StringComparison.OrdinalIgnoreCase))
+                    {
+                        continue;
+                    }
+
+                    if (currentKey == null || mapping.Key.Length >= currentKey.Length)
+                    {
+                        currentKey = mapping.Key;
+                        mappedTableName = mapping.Value;
+                    }
+                }
+            }
+
+            mappedTableName ??= !this.m_shouldPassThruTableMappings
+                ? this.m_defaultTableName
+                : s_passthroughTableName;
+
+            if (mappedTableName.Length <= 0 && tableNameCache.CachedSanitizedTableNameCount < MaxCachedSanitizedTableNames)
+            {
+                // We resolved to a wildcard which is pass-through mode.
+
+                Span<byte> stackBuffer = stackalloc byte[64];
+
+                int bytesWritten = SanitizedCategoryName(stackBuffer, categoryName);
+                if (bytesWritten > 0)
+                {
+                    mappedTableName = stackBuffer.Slice(0, bytesWritten).ToArray();
+
+                    tableNameCache.CachedSanitizedTableNameCount++;
+                }
+                else
+                {
+                    // Note: When the table name could not be sanitized we cache
+                    // the empty array NOT s_passthroughTableName.
+                    mappedTableName = Array.Empty<byte>();
+                }
+            }
+
+            // Note: This is using copy-on-write pattern to keep the happy
+            // path lockless once everything has spun up.
+            this.m_tableNameCache = new(tableNameCache)
+            {
+                [categoryName] = mappedTableName,
+            };
+
+            return mappedTableName;
+        }
+    }
+
+    // Note: This is used for tests.
+    public interface ITableNameCacheDictionary : IReadOnlyDictionary<string, byte[]>
+    {
+        int CachedSanitizedTableNameCount { get; }
+    }
+
+    private sealed class TableNameCacheDictionary : Dictionary<string, byte[]>, ITableNameCacheDictionary
+    {
+        public TableNameCacheDictionary()
+            : base(0, StringComparer.OrdinalIgnoreCase)
+        {
+        }
+
+        public TableNameCacheDictionary(TableNameCacheDictionary sourceCache)
+            : base(sourceCache, StringComparer.OrdinalIgnoreCase)
```

Comment:
`TableNameMappings` uses `StringComparer.Ordinal`.

Reply:
Thanks, didn't notice that. Looked into it a bit. I was going to change to `OrdinalIgnoreCase` because it seems low-risk and nice for users but looking at the code `Ordinal` is used for all the lookup dictionaries. I'm going to set as `Ordinal` for now and look at that as a follow-up. I did refactor a bit so it would be easier to change in the future.